storage/remote/storage.go

```diff
@@ -62,42 +63,53 @@ func (s *Storage) ApplyConfig(conf *config.Config) error {
 
 	// Update write queues
 
-	newQueues := []*QueueManager{}
-	// TODO: we should only stop & recreate queues which have changes,
-	// as this can be quite disruptive.
+	newQueues := make([]*QueueManager, 0, len(s.queues))
+	oldQueues := s.queues
+
 	for i, rwConf := range conf.RemoteWriteConfigs {
-		c, err := NewClient(i, &ClientConfig{
-			URL:              rwConf.URL,
-			Timeout:          rwConf.RemoteTimeout,
-			HTTPClientConfig: rwConf.HTTPClientConfig,
-		})
+		ok := i < len(oldQueues)
```

Comment:
Why introduce an extra variable for this instead of just saying `if i < len(oldQueues)`?